pkg/autodiscovery/providers/kubecontainer.go

```diff
@@ -163,19 +164,21 @@ func (k *KubeContainerConfigProvider) generateConfig(e workloadmeta.Entity) ([]i
 
 	switch entity := e.(type) {
 	case *workloadmeta.Container:
-		containerID := entity.GetID().ID
-		containerEntityName := containers.BuildEntityName(string(entity.Runtime), containerID)
-		configs, errs = utils.ExtractTemplatesFromContainerLabels(containerEntityName, entity.Labels)
-
-		// AddContainerCollectAllConfigs is only needed when handling
-		// the container event, even when the container belongs to a
-		// pod. Calling it when handling the KubernetesPod will always
-		// result in a duplicated config, as each KubernetesPod will
-		// also generate events for each of its containers.
-		configs = utils.AddContainerCollectAllConfigs(configs, containerEntityName)
-
-		for idx := range configs {
-			configs[idx].Source = names.Container + ":" + containerEntityName
+		// kubernetes containers need to be handled together with their
+		// pod, so they generate a single []integration.Config.
+		// otherwise, with container_collect_all, it's possible for a
+		// container that belongs to an AD-annotated pod to briefly
+		// have a container_collect_all when it shouldn't.
```

Comment:
```suggestion
		// otherwise, it's possible for a container that belongs
		// to an AD-annotated pod to briefly be scheduled without its
		// annotations.
```